### PR TITLE
feat!: remove gherkin, add events support

### PR DESCRIPTION
**Pull request by toddbaert**
This PR adds a means of testing events to the [flagd-testbed](https://github.com/open-feature/test-harness/pkgs/container/flagd-testbed) container image.

* removes the gherkin suite (which is now in the spec)
* removes the caching suite (it was no longer implemented anywhere)
* adds a small shell script and a new flag that changes once a second, to easily test flag change events

### CODE REVIEW

File: wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
```
Comment (toddbaert) [inline wrapper.sh:1]: We need this little wrapper script to start our `flags/change-flag.sh` script in the background, and to forward signals to flagd so we can gracefully shut down.

File: flags/change-flag.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
```
Comment (toddbaert) [inline flags/change-flag.sh:1]: This script just swaps the value of the flag in `changing-flag.json` once a second.

File: Dockerfile [hunk 3]
```diff
@@ -1,6 +1,10 @@
-FROM ghcr.io/open-feature/flagd:v0.5.2
+FROM ghcr.io/open-feature/flagd:v0.6.3 as flagd
 
-COPY testing-flags.json testing-flags.json
+FROM busybox:1.36
```
Comment (toddbaert) [inline Dockerfile:3]: busybox because we need some basic tools and a shell.

File: flags/testing-flags.json
Comment (Kavindu-Dodan) [inline flags/testing-flags.json:1]: minor - we can git rid of the symlink file here : https://github.com/open-feature/test-harness/blob/main/symlink_testing-flags.json

File: wrapper.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+handle_int() { 
+  kill -INT "$child" 2>/dev/null
+}
+
+trap handle_term SIGTERM
+trap handle_int SIGINT
+
+# start change scriptm and flagd
```
Comment (Kavindu-Dodan) [inline wrapper.sh:16]:
```suggestion
# start change script and flagd
```

**Review by Kavindu-Dodan — APPROVED** [review]
Nice work 🙌  

Added few nits and looks good

File: Dockerfile (hunk 3 above)
Comment (Kavindu-Dodan) [reply Dockerfile:3]: This is fine. Final image is ~60MB

File: flags/testing-flags.json
Comment (toddbaert) [reply flags/testing-flags.json:1]: Your right.

File: features/evaluation.feature [hunk 5]
```diff
@@ -1,67 +0,0 @@
-Feature: Flag evaluation
```
Comment (Kavindu-Dodan) [inline features/evaluation.feature:1]: we should not remove this yet. there are other references for these definitions. for example - https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/pom.xml#L211-L212

**Review by Kavindu-Dodan — CHANGES_REQUESTED** [review]
I am not confident about gherkin test removal. There are other projects with submodule references for these

File: features/evaluation.feature (hunk 5 above)
Comment (toddbaert) [reply features/evaluation.feature:1]: They are bound to old releases, not main. This is breaking, but it won't suddenly break any dependency repos, they are bound to release tags.

**Comment by toddbaert** [comment]
> I am not confident about gherkin test removal. There are other projects with submodule references for these

https://github.com/open-feature/test-harness/pull/51#discussion_r1309511447

File: features/evaluation.feature (hunk 5 above)
Comment (Kavindu-Dodan) [reply features/evaluation.feature:1]: 🤔  hmm, then we should avoid merging renovate PRs of dependent repositories

Comment (Kavindu-Dodan) [reply features/evaluation.feature:1]: 🤔  then we should avoid updating submodules on dependent repositories

Comment (toddbaert) [reply features/evaluation.feature:1]: Yes those repositories should instead use the spec now, which has these gherken suites. I have a WIP PR for the js-sdk-contribs.

**Review by beeme1mr — APPROVED** [review]

**Review by Kavindu-Dodan — APPROVED** [review]

**Pull request merged**